tests/cpgrid/coord_zcorn_lgr_test.cpp

```diff
@@ -0,0 +1,277 @@
+
+    for (int level = 0; level < grid.maxLevel()+1; ++level) {
+        for (const auto& element : elements(grid.levelGridView(level))) {
+            const auto& global_cell_level = grid.currentData()[element.level()]->globalCell()[element.index()];
+            if (element.isLeaf()) {
+                const auto& leaf_idx = localCartesianIdxSets_to_leafIdx[element.level()].at(global_cell_level);
+                BOOST_CHECK_EQUAL(leafIdx_to_localCartesianIdxSets[leaf_idx][0], element.level());
+                BOOST_CHECK_EQUAL(leafIdx_to_localCartesianIdxSets[leaf_idx][1], global_cell_level);
+            } else {
+                BOOST_CHECK_THROW(localCartesianIdxSets_to_leafIdx[element.level()].at(global_cell_level),
+                                  std::out_of_range);
+            }
+        }
+    }
+
+
+    for (int level = 1; level < grid.maxLevel()+1; ++level) {
+        // Get COORD values for LGRs
+        // COORD defines a set of coordinate lines or pillars for a reservoir grid via an array.
+        // A total of 6 x (NX+1) x (NY+1) lines must be specified for each coordinate data set (or reservoir).
+        // For Cartesian geometry, each line is defined by the (x, y, z) coordinates of two distinct points on the line. The
+        // lines are entered with I cycling fastest then J.
+        //
+        // For each LGR, we collect the coordinates of the vertices. Attention: pillar ordering not guaranteed.
+        std::vector<std::array<double,3>> xyz_coord(grid.currentData()[level]->size(3)); // NOT ORDERED IN A 'PILLAR' WAY
+        for(const auto& point : vertices(grid.levelGridView(level))) {
+            for(int i = 0; i<3; ++i) {
+                xyz_coord[point.index()][i] = point.geometry().center()[i];
+            }
+            std::cout<< "level: " << level << " x: " << xyz_coord[point.index()][0] << " y: "<< xyz_coord[point.index()][1] << " z: "<< xyz_coord[point.index()][2] << std::endl;
+        }
+
+        // Get ZCORN values for LGRs
+        // ZCORN defines the depth of each corner point of a grid block on the pillars defining the reservoir grid. A
+        // total of 8 x NX x NY x NZ values are needed to fully define all the depths in the model. The depths
+        // specifying the top of the first layer are entered first with one point for each pillar for each grid block. The
+        // points are entered with the X axis cycling fastest. Next come the depths of the bottom of the first layer. The
+        // top of layer two follows etc.
+        //
+        // Expected values in each LGR: 8*refined_cells_in_that_lgr
+        // Each refined cell has its 8 corners stored in <refined_level_grid>.cell_to_point_[ refined element index ]
+        // in the following order:
+        //     BOTTOM FACE          TOP FACE
+        //     2 --- 3              6 --- 7
+        //    /     /              /     /
+        //   0 --- 1              4 --- 5
```

Comment:
It typically does not matter, but for what it's worth, the default coordinate system is right-handed with the origin in the upper, left, back corner and Z axis pointing down.  In other words, the **typical** node ordering is as outlined in the picture in MRST's [`processGRDECL`](https://github.com/SINTEF-AppliedCompSci/MRST/blob/5bcfe7cd64c2a9c3bdb298af46c7fe4c0df8feb9/core/gridprocessing/processGRDECL.m#L136-L149) function.

Reply:
I see, in the grid, we follow the described ordering. Therefore, for output, we need to re-order them according to your description. Thanks for your feedback!

Reply:
> Therefore, for output, we need to re-order them according to your description.

Sorry for being too brief: It's not that we need to do anything in particular to the order&ndash;it's actually the same in both views&ndash;it's just that in terms of labelling the **top** face is normally comprised of vertices 0..3 while the **bottom** face normally comprises vertices 4..7.

Reply:
Thanks for clarifying! I added your comment into the documentation.